lib/cdo/crowdin/client_extentions.rb

```diff
@@ -0,0 +1,151 @@
+require 'crowdin-api'
+
+#
+# Add functionalities to the Crowdin::Client class in the crowdin-api gem.
+#
+module Crowdin
+  module ClientExtensions
+    # Project Id is at https://crowdin.com/project/<project_name>/tools/api
+    # Project source language is at https://crowdin.com/project/<project_name>/settings
+    CDO_PROJECTS = {
+      codeorg: {
+        id: 26074,
+        source_language: 'enus'
+      },
+      'hour-of-code': {
+        id: 55536,
+        source_language: 'en'
+      },
+      'codeorg-markdown': {
+        id: 314545,
+        source_language: 'en'
+      },
+      'codeorg-restricted': {
+        id: 464582,
+        source_language: 'en'
+      }
+    }
+
+    # Maximum number of items to retrieve from Crowdin in an API call
+    MAX_ITEMS_COUNT = 500
+
+    # Send requests to Crowdin in a loop until the number of items retrieved
+    # is less than an expected limit.
+    #
+    # @param query [Hash] query string params used in the requests
+    # @yieldparam query [Hash] updated query string params
+    # @yieldreturn [Array] the response from Crowdin
+    # @return [Array] all responses from Crowdin
+    #
+    def request_loop(query)
+      raise 'Missing block' unless block_given?
+
+      results = []
+      loop do
+        # yield to the given block to do real work
+        response = yield query
+        raise 'Crowdin request failed!' if response.is_a?(String) && response.match?('Something went wrong')
+
+        puts "Offset = #{query[:offset]}. Response size = #{response['data'].size}"
+        results.concat(response['data'])
+        query[:offset] += response['data'].size
+        break if response['data'].size < query[:limit]
```

Comment:
Though unlikely, what would happen if the number of results divides evenly into 500, and the size of the final response == 500?